src/Simplex/Chat.hs

```diff
@@ -4260,6 +4289,23 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
           groupMsgToView g' m ci msgMeta
         createGroupFeatureChangedItems user cd CIRcvGroupFeature g g'
 
+    directMsgReceived :: Contact -> Connection -> MsgMeta -> NonEmpty MsgReceipt -> m ()
+    directMsgReceived ct@Contact {contactId} Connection {connId} msgMeta msgRcpts = do
+      checkIntegrityCreateItem (CDDirectRcv ct) msgMeta
+      forM_ msgRcpts $ \MsgReceipt {agentMsgId, msgRcptStatus} -> do
+        withStore $ \db -> createSndMsgDeliveryEvent db connId agentMsgId $ MDSSndRcvd msgRcptStatus
+        withStore' (\db -> getDirectChatItemByAgentMsgId db user contactId connId agentMsgId) >>= \case
+          Just (CChatItem SMDSnd ci) -> do
+            chatItem <- withStore $ \db -> updateDirectChatItemStatus db user contactId (chatItemId' ci) $ CISSndRcvd msgRcptStatus
+            toView $ CRChatItemReceipt user msgRcptStatus $ AChatItem SCTDirect SMDSnd (DirectChat ct) chatItem
```

Comment:
we have CRChatItemStatusUpdated for SENT, could it be reused here?

Reply:
that status even already has MsgReceiptStatus

Reply:
possibly, I wanted to differentiate the output for tests, but probably it can be done there